controllers/tasks.js

```diff
@@ -128,16 +128,24 @@ const fetchPaginatedTasks = async (query) => {
 
 const fetchTasks = async (req, res) => {
   try {
-    const { dev, status, page, size, prev, next } = req.query;
+    const { dev, status, page, size, prev, next, q: queryString } = req.query;
     const transformedQuery = transformQuery(dev, status, size, page);
-
+    const searchTerm = queryString || "";
     if (dev) {
       const paginatedTasks = await fetchPaginatedTasks({ ...transformedQuery, prev, next });
       return res.json({
         message: "Tasks returned successfully!",
         ...paginatedTasks,
       });
     }
+    if (queryString) {
+      const filterTasks = await tasks.fetchTasks(searchTerm);
+      const tasksWithRdsAssigneeInfo = await fetchTasksWithRdsAssigneeInfo(filterTasks);
+      return res.json({
+        message: "Filter tasks returned successfully!",
+        tasks: tasksWithRdsAssigneeInfo.length > 0 ? tasksWithRdsAssigneeInfo : [],
```

Comment:
if the array is empty the message can be updated to `no tasks found` instead of `Filter tasks returned successfully`

Reply:
done